scripts/building_placer.gd

```diff
@@ -3,19 +3,14 @@ extends Node3D
 @export var max_number_buildings_on_x_axis : int = 20
 @export var max_number_buildings_on_z_axis : int = 20
 @export var distance_between_buildings_metres : int = 20
-@export var inherited_offset_correction : int = -220
-
-# Toggle to stop the physics process fuction repeating the building placement task
-var rays_fired : int = 0
+@export var inherited_offset_correction : int = 0
 
 var building001 = preload("res://Scenes/Buildings/Building001.tscn")
 
-# Called when the node enters the scene tree for the first time.
 func _ready():
 	pass
 
-func _physics_process(_delta):
-	if rays_fired == 0:
+func _on_ditzy_terrain_landscape_complete():
 		for count_building_on_x_axis in max_number_buildings_on_x_axis:
 			for count_building_on_z_axis in max_number_buildings_on_z_axis:
 				var x_location : float = ((0 - (max_number_buildings_on_x_axis / 2.0) + count_building_on_x_axis) * distance_between_buildings_metres) - inherited_offset_correction
```

Comment:
[style]
There is a lot of indentation here and some very long lines which pushes a lot of the code off the right-hand edge of the editor. I generally try to keep code within the 100-character guide (the second line in the editor), or I may go over a little but I'd certainly be looking to refactor if a line goes over 120 characters.

There are three things I'd try in this case: 1) remove the unnecessary extra level of indentation at the start of the function, 2) break out the nested part into it's own function, and 3) split long lines with a `\` character which prevents godot from treating the line-break as the end of the statement so you can write more on the next line(s).

i.e. 
```
func _on_ditzy_terrain_landscape_complete():
	for count_building_on_x_axis in max_number_buildings_on_x_axis:
		for count_building_on_z_axis in max_number_buildings_on_z_axis:
			_place_building(count_building_on_x_axis, count_building_on_z_axis)

func _place_building(count_building_on_x_axis, count_building_on_z_axis):
	var x_location : float = ((count_building_on_x_axis - (max_number_buildings_on_x_axis / 2.0)) \
		* distance_between_buildings_metres) - inherited_offset_correction
# etc...
```

I also changed the order of operations a little because `0 - A + B` is the same as `B - A`, which I think is a bit more concise and easier to read.

Reply:
Now that I've submitted the review, I notice that even my shortened version still has a scrollbar here on github meaning you have to move it to the right if you want to see the last few characters (though compare to your code in the upper part of this review comment where you have to scroll left-and-right a lot to take in the code!)

Anyway, it should perhaps be even shorter than in my initial suggestion, so maybe:
```
func _place_building(count_building_on_x_axis, count_building_on_z_axis):
	var x_location : float = \
		((count_building_on_x_axis - (max_number_buildings_on_x_axis / 2.0)) \
		* distance_between_buildings_metres) - inherited_offset_correction
```